cli/loader_win_utils.c

```diff
@@ -12,22 +12,30 @@ static FILE _stderr = { INVALID_HANDLE_VALUE };
 FILE *stdout = &_stdout;
 FILE *stderr = &_stderr;
 
-int loader_fwrite(const WCHAR *str, size_t nchars, FILE *out) {
+int loader_fwrite(const char *str, size_t nchars, FILE *out) {
     DWORD written;
     if (out->isconsole) {
-        if (WriteConsole(out->fd, str, nchars, &written, NULL))
+        // Windows consoles do not support UTF8, only UTF16.
+        size_t wbufsz = nchars * 4 + 1;
```

Comment:
```suggestion
        size_t wbufsz = nchars * 3 + 1;
```

Reply:
I may have overshot.

I'll check if there any unicode codepoints that take two or less code units to express in utf8, but the full two code units to express in utf16. If the answer is no, then `nchars * 2 + 1` will suffice. If the answer is yes, then it must be more. `nchars * 4 + 1` is "definitely safe," as it assumes that byte will fill two utf16 code units, like you would with utf32.

If the answer is somewhere inbetween, I'm not quite sure where. But I chose x4 to be safe.

Reply:
U+0000 to U+D7FF encode as a single utf-16 codepoint (which is all 1, 2, and some 3 byte utf8 encodings), and otherwise as 2 codepoints (for 3 and 4 byte utf8), so `*2` suffices

But a better solution would be to change utf8_to_wchar and wchar_to_utf8 to return new malloc memory. Could you do that as a followup PR, removing as many of the statically-sized buffers as possible in this file?